src/test/java/com/jos/dem/vetlog/service/PetServiceTest.java

```diff
@@ -214,4 +218,17 @@ void shouldNotDeletePetById(TestInfo testInfo) {
             service.deletePetById(1L);
         });
     }
+
+    @Test
+    @DisplayName("not deleting a pet due to IN_ADOPTION status")
+    void shouldNotDeletePetInAdoption(TestInfo testInfo) {
+        // given
+        log.info("Running: {}", testInfo.getDisplayName());
+        var petInAdoption = new Pet();
+        petInAdoption.setStatus(PetStatus.IN_ADOPTION);
+        Mockito.when(petRepository.findById(1L)).thenReturn(Optional.of(petInAdoption));
+
```

Comment:
Maybe we want to be uniform in the mockito usage, please see line 196:
`when(adoptionRepository.findByPet(pet)).thenReturn(optional);`

Also in assertions, we can do:
`assertThrows(BusinessException.class, () -> service.deletePetById(1L));`